Dst io

jjgomezcadenas opened the pull request:
This PR introduces a protection in the function that reads dorothea DSTs, throwing an exception if the operation is not successful. This protection is necessary to prevent the case in which a spurious file appears in a directory containing DSTs (and being read out using glob). Without the exception, the function will read files until it finds the wrong file, then die ungracefully, and the concatenation of files will not occur.

jjgomezcadenas committed:
Throw an exception if the table is not correctly read

jjgomezcadenas committed:
Add an excecption to describe a DST table not correctly read.

jjgomezcadenas committed:
Add a "wrong file" in the return of KrMC_kdst fixture.

This is needed to test the case in which one accidentally reads a
wrong file. The case occurs, for example, if one is reading dst files
with glob, and a file of different type than expected is present in
the directory.

jjgomezcadenas committed:
Adapt test, KrMC_kdst is now a tuple.

jjgomezcadenas committed:
Add a file of "wrong type" to test reading DSTs

jacg commented on invisible_cities/io/dst_io.py:
```diff
@@ -1,13 +1,29 @@
-        table = getattr(getattr(h5in.root, group), node).read()
+        try:
+            table = getattr(getattr(h5in.root, group), node).read()
+        except:
```
`except` without an exception specification is rarely the correct thing to do. You are saying, "I don't care what message is being sent here, whatever it is I know that the way to deal with it is to print a message to standard out and to translate the exception to `UnknownDST`.

What if the message is "Incoming nuclear missile, run to the nuclear shelters right now". Who cares, let's print a message to standard out and translate it to `UnknownDST`.

Please handle the exceptions that you are expecting, and let all others propagate down the stack to whoever *really* knows how to deal with them.

jjgomezcadenas committed:
Add assert to test.

jjgomezcadenas committed:
Remove commented code

The proposed behavior of the function when an exception of type
NoSuchNodeError is found is simply to print an error message, to
inform the user of the existence of an offending file and then ignore
the file. This is a case where it is justified not to abort the concat
of good files (which can take many minutes) because a spurious or
corrupted file is found. No need to throw a new exception either,
although it could be done if one wanted to let the user decide what to do.

jjgomezcadenas committed:
Replace print by warning.

jmbenlloch reviewed (approved):
Simple update including a test.
